### PR TITLE
WIP: [Fix_bug_no_64] Fix bug 1. Compare time with !=, <> 2. IN/NOT IN with all type

### PR DESCRIPTION
**Pull request by khieuvm**
1. Influxdb not support compare time column with != or <>
--> Not pushdown compare time column with != or <>
2. IN/NOT IN parsing error:
currently IN/NOT IN parsing lost some case
--> Add more case, fix case BOOL parsing.

### CODE REVIEW

File: sql/influxdb_fdw.sql [hunk 1]
```diff
@@ -52,6 +52,10 @@ SELECT * FROM cpu WHERE value3 IS NULL;
 SELECT * FROM cpu WHERE tag2 IS NULL;
 SELECT * FROM cpu WHERE value3 IS NOT NULL;
 SELECT * FROM cpu WHERE tag2 IS NOT NULL;
+
+-- influx not support pushdown compare timestamp with != or <>
+SELECT * FROM cpu WHERE time != '2015-08-18 09:48:08+09';
```
Comment (binhnv56) [inline sql/influxdb_fdw.sql:57]: I see other test case will compare time ti TIMESTAMP (WHERE time = TIMESTAMP), so I think you should add test case for compare time != TIMESTAMP to improve the quality

File: deparse.c [hunk 2]
```diff
@@ -476,6 +476,31 @@ foreign_expr_walker(Node *node,
...
+				{
+					Var *var;
+					ListCell *lc;
+
+					/* Check Timestamp type for the operand which is Var */
+					foreach(lc, oe->args)
+					{
+						var = lfirst(lc);
+						if (!IsA(var, Var))
+							continue;
```
Comment (thaiphanvan0779) [inline deparse.c:500]: lfirst(lc) is an address can point to anything, you should cast to an Expr pointer and then check whether it is a Var node or not

File: deparse.c [hunk 3]
```diff
@@ -1852,9 +1881,22 @@ influxdb_deparse_scalar_array_op_expr(ScalarArrayOpExpr *node, deparse_expr_cxt
 	getTypeOutputInfo(c->consttype,
 					  &typoutput, &typIsVarlena);
 	extval = OidOutputFunctionCall(typoutput, c->constvalue);
-	isstr = true;
-	if (c->consttype == INT4ARRAYOID || c->consttype == OIDARRAYOID)
-		isstr = false;
+	
+	switch (c->consttype)
+	{
+		case BOOLARRAYOID:
```
Comment (thaiphanvan0779) [inline deparse.c:1887]: In line of 143, you define BOOLARRAYIOD in #if (PG_VERSION_NUM < 110000)
It should if #if... #endif here

File: deparse.c [hunk 4]
```diff
@@ -1913,6 +1955,17 @@ influxdb_deparse_scalar_array_op_expr(ScalarArrayOpExpr *node, deparse_expr_cxt
 			deparseLeft = true;
 			continue;
 		}
+
+		/* Influx only support "= true" or "= false" */
+		if (c->consttype == BOOLARRAYOID){
```
Comment (thaiphanvan0779) [inline deparse.c:1960]: In line of 143, you define BOOLARRAYIOD in #if (PG_VERSION_NUM < 110000)
It should if #if... #endif here

Comment (khieuvm) [reply deparse.c:1968]: I fixed it.

File: deparse.c (hunk 3 above)
Comment (khieuvm) [reply deparse.c:1895]: I fixed it.

Comment (khieuvm) [reply deparse.c:1895]: I fixed it

File: deparse.c (hunk 4 above)
Comment (khieuvm) [reply deparse.c:1968]: I fixed it

File: deparse.c (hunk 2 above)
Comment (khieuvm) [reply deparse.c]: I fixed it

**Comment by thaiphanvan0779** [comment]
I have no more comment

**Pull request closed without merging**